ament_yamllint/ament_yamllint/main.py

```diff
@@ -0,0 +1,275 @@
+# limitations under the License.
+
+import argparse
+import errno
+import os
+import re
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+from xml.sax.saxutils import escape
+from xml.sax.saxutils import quoteattr
+
+import yaml
+
+
+def main(argv=sys.argv[1:]):
+    config_file = os.path.join(
+        os.path.dirname(__file__), 'configuration', 'yamllint.yaml')
+
+    extensions = ['yaml', 'yml']
+
+    parser = argparse.ArgumentParser(
+        description='Check YAML style using YAMLlint.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        '-c', '--config',
+        metavar='CFG',
+        default=config_file,
+        dest='config_file',
+        help='The config file')
+    parser.add_argument(
+        '--linelength', metavar='N', type=int,
+        help='The maximum line length (default: specified in the config file)')
+    parser.add_argument(
+        'paths',
+        nargs='*',
+        default=[os.curdir],
+        help='The files or directories to check. For directories files ending '
+             'in %s will be considered.' %
+             ', '.join(["'.%s'" % e for e in extensions]))
+    parser.add_argument(
+        '--exclude',
+        nargs='*',
+        default=[],
+        help='Exclude specific file names and directory names from the check')
+    # not using a file handle directly
+    # in order to prevent leaving an empty file when something fails early
+    parser.add_argument(
+        '--xunit-file',
+        help='Generate a xunit compliant XML file')
+    args = parser.parse_args(argv)
+
+    if not os.path.exists(args.config_file):
+        print("Could not config file '%s'" % args.config_file, file=sys.stderr)
+        return 1
+
+    files = get_files(args.paths, extensions, args.exclude)
+    if not files:
+        print('No files found', file=sys.stderr)
+        return 1
+
+    if args.xunit_file:
+        start_time = time.time()
+
+    with open(args.config_file) as f:
+        yamllint_config = yaml.safe_load(f)
+
+    assert isinstance(yamllint_config, dict), 'Invalid configuration file'
+    yamllint_config['yaml-files'] = ['*.%s' % (ext,) for ext in extensions]
+    if args.linelength is not None:
+        if not isinstance(yamllint_config.get('rules'), dict):
+            yamllint_config['rules'] = {}
+        if not isinstance(yamllint_config['rules'].get('line-length'), dict):
+            yamllint_config['rules']['line-length'] = {}
+        yamllint_config['rules']['line-length']['max'] = args.linelength
+
+    temp_config_fd, temp_config_file = tempfile.mkstemp(suffix='.yaml',
+                                                        prefix='yamllint_')
+    with os.fdopen(temp_config_fd, 'w') as f:
+        yaml.dump(yamllint_config, f)
+
+    try:
+        report = invoke_yamllint(files, config_file=temp_config_file)
+    except:  # noqa: E722
+        raise
+    finally:
+        os.remove(temp_config_file)
```

Comment:
I think this whole thing might be better done with a [NamedTemporaryFile](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) in a context.  That way we are always sure it is removed, even if `yaml.dump`, for instance, throws an exception.